src/pages/images/IsoSelector.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC } from "react";
+import { Button, MainTable } from "@canonical/react-components";
+import { humanFileSize, isoTimeToString } from "util/helpers";
+import { useQuery } from "@tanstack/react-query";
+import { loadIsoImages } from "context/loadIsoImages";
+import { queryKeys } from "util/queryKeys";
+import Loader from "components/Loader";
+import { useProject } from "context/project";
+import { RemoteImage } from "types/image";
+import { IsoImage } from "pages/images/ImageSelectorModal";
+
+interface Props {
+  primaryImage: IsoImage | null;
+  onSelect: (image: RemoteImage, type: string | null) => void;
+  onUpload: () => void;
+  onCancel: () => void;
+}
+
+const CustomIsoSelector: FC<Props> = ({
+  primaryImage,
+  onSelect,
+  onUpload,
+  onCancel,
+}) => {
+  const { project } = useProject();
+  const projectName = project?.name ?? "";
+
+  const { data: images = [], isLoading } = useQuery({
+    queryKey: [queryKeys.isoImages, project],
+    queryFn: () => loadIsoImages(projectName),
+  });
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Storage pool", sortKey: "storagePool" },
+    { content: "Upload date", sortKey: "uploadedAt" },
+    { content: "Size", sortKey: "size" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = images.map((image) => {
+    return {
+      columns: [
+        {
+          content: image.aliases,
+          role: "cell",
+          "aria-label": "Name",
+        },
+        {
+          content: image.pool,
+          role: "cell",
+          "aria-label": "Storage pool",
+        },
+        {
+          content: isoTimeToString(new Date(image.created_at).toISOString()),
+          role: "cell",
+          "aria-label": "Uploaded at",
+        },
+        {
+          content:
+            image.volume?.config.size &&
+            humanFileSize(+image.volume.config.size),
+          role: "cell",
+          "aria-label": "Size",
+        },
+        {
+          content: (
+            <Button
+              appearance={
+                primaryImage?.name === image.aliases &&
+                primaryImage.pool === image.pool
+                  ? "positive"
+                  : ""
+              }
+              onClick={() => onSelect(image, "virtual-machine")}
+              dense
+            >
+              Select
+            </Button>
+          ),
```

Comment:
![Screenshot from 2023-09-06 09:38:26](https://github.com/canonical/lxd-ui/assets/56583786/116452d4-54bd-413b-ba89-60c12d8496d6)

My gut feeling tells me that users would appreciate a delete button to get rid of ISOs much more here than where it is now (hidden in the storage pool detail page, volumes tab). I suggest to have it in both places: we could add just a delete icon at the end of the row, after the "Select" button.